java/client/src/main/java/redis/clients/jedis/ConfigurationMapper.java

```diff
@@ -77,12 +290,136 @@ public static GlideClientConfiguration mapToGlideConfig(
+        int blockingTimeout = jedisConfig.getBlockingSocketTimeoutMillis();
+
+        // Use the maximum of connection and socket timeouts
+        int maxTimeout = Math.max(connectionTimeout, socketTimeout);
+
+        // If blocking timeout is specified and reasonable, consider it
+        if (blockingTimeout > 0 && blockingTimeout < 300000) { // Less than 5 minutes
+            maxTimeout = Math.max(maxTimeout, blockingTimeout);
+        }
+
+        // Ensure minimum timeout
+        return Math.max(maxTimeout, 1000); // At least 1 second
+    }
+
+    /** NEW: Validates that Jedis configuration can be converted to GLIDE. */
+    public static ValidationResult validateConfiguration(JedisClientConfig jedisConfig) {
+        List<String> warnings = new ArrayList<>();
+        List<String> errors = new ArrayList<>();
+
+        // Check for unsupported features
+        if (jedisConfig.getAuthXManager() != null) {
+            errors.add("AuthXManager is not supported in GLIDE");
+        }
+
+        // Check SSL configuration
+        if (jedisConfig.isSsl()) {
+            validateSslConfiguration(jedisConfig, warnings, errors);
+        }
+
+        // Check timeout configurations
+        validateTimeoutConfiguration(jedisConfig, warnings);
+
+        // Check database selection
+        if (jedisConfig.getDatabase() != Protocol.DEFAULT_DATABASE) {
+            warnings.add("Database selection: GLIDE supports database selection for standalone mode");
+        }
+
+        return new ValidationResult(warnings, errors);
+    }
+
+    /** Validates SSL configuration for conversion compatibility. */
+    private static void validateSslConfiguration(
+            JedisClientConfig jedisConfig, List<String> warnings, List<String> errors) {
+
+        if (jedisConfig.getSslSocketFactory() != null) {
+            errors.add("Custom SSLSocketFactory is not supported in GLIDE");
+        }
+
+        if (jedisConfig.getHostnameVerifier() != null) {
+            errors.add("Custom HostnameVerifier is not supported in GLIDE");
+        }
+
+        // Check SSL parameters for unsupported configurations
+        if (jedisConfig.getSslParameters() != null) {
+            SSLParameters sslParams = jedisConfig.getSslParameters();
+            if (sslParams.getEndpointIdentificationAlgorithm() != null
+                    && sslParams.getEndpointIdentificationAlgorithm().isEmpty()) {
+                errors.add("Disabled endpoint identification is not supported in GLIDE");
+            }
+        }
+
+        SslOptions sslOptions = jedisConfig.getSslOptions();
+        if (sslOptions != null) {
+            if (sslOptions.getSslVerifyMode() == SslVerifyMode.INSECURE) {
+                warnings.add(
+                        "SSL verification disabled via SslVerifyMode.INSECURE. Will map to GLIDE's"
+                                + " useInsecureTLS option.");
+            }
+
+            // Check if certificate resources are accessible
```

Comment:
Why do we just test accessibility? This option doesn't translate to GLIDE. This is a potential security risk because the user will think they are using this keystore when they aren't.